WcaOnRails/app/webpacker/components/Delegates/index.jsx

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React from 'react';
+
+import { Button } from 'semantic-ui-react';
+import I18n from '../../lib/i18n';
+import UserBadge from '../UserBadge';
+
+import '../../stylesheets/delegates/style.scss';
+
+const dasherize = (string) => string.replace(/_/g, '-');
+
+export default function Delegates({
+  delegates,
+  isEditVisible,
+}) {
+  const getSeniorDelegates = () => delegates
+    .filter((user) => user.delegate_status === 'senior_delegate')
+    .sort((user1, user2) => {
+      if (user1.region < user2.region) {
+        return -1;
+      } if (user1.region > user2.region) {
+        return 1;
+      }
+      return 0;
+    });
+
+  const [seniorDelegates] = React.useState(getSeniorDelegates());
```

Comment:
As written, the initial state `getSeniorDelegates()` for this state will be recalculated every render, even though it's only used once (the initial render). If you pass in the literal function `getSeniorDelegates` instead of the result `getSeniorDelegates()`, it will only calculate it the first render. [See the docs](https://legacy.reactjs.org/docs/hooks-reference.html#lazy-initial-state).

Having said that, if you're not using a setter then you don't need a state. This could be refactored to
```
const seniorDelegates = delegates
    .filter((user) => user.delegate_status === 'senior_delegate')
    .sort((user1, user2) => {
      if (user1.region < user2.region) {
        return -1;
      } if (user1.region > user2.region) {
        return 1;
      }
      return 0;
    });
```
or
```
const seniorDelegates = useMemo(() => delegates
    .filter((user) => user.delegate_status === 'senior_delegate')
    .sort((user1, user2) => {
      if (user1.region < user2.region) {
        return -1;
      } if (user1.region > user2.region) {
        return 1;
      }
      return 0;
    }), [delegates]);
```

Reply:
Done, thank you.